src/contexts/bridge.context.tsx

```diff
@@ -130,68 +154,114 @@ const BridgeProvider: FC = (props) => {
     [env]
   );
 
-  const bridge = useCallback(
-    async (
-      token: string,
-      amount: BigNumber,
-      destinationNetwork: number,
-      destinationAddress: string
-    ): Promise<ContractTransaction> => {
+  const estimateBridgeGasPrice = useCallback(
+    ({ from, token, to, destinationAddress }: EstimateBridgeGasPriceParams) => {
       if (env === undefined) {
-        throw new Error("Environment is not available");
+        throw new Error("Env is not available");
       }
 
-      if (bridgeContract === undefined) {
+      const amount = parseUnits("1", token.address);
+      const contract =
+        from.name === "ethereum"
+          ? Bridge__factory.connect(env.bridge.l1ContractAddress, from.provider)
+          : Bridge__factory.connect(env.bridge.l2ContractAddress, from.provider);
+      const overrides: PayableOverrides =
+        token.address === ethersConstants.AddressZero ? { value: amount } : {};
+
+      if (contract === undefined) {
         throw new Error("Bridge contract is not available");
       }
 
-      if (token === ethersConstants.AddressZero) {
-        return bridgeContract.bridge(token, amount, destinationNetwork, destinationAddress, {
-          value: amount,
+      return contract.estimateGas
+        .bridge(token.address, amount, to.bridgeNetworkId, destinationAddress, {
+          ...overrides,
+          from: destinationAddress,
+        })
+        .then((gasLimit) => {
+          const gasIncrease = gasLimit.div(BRIDGE_CALL_GAS_INCREASE).toNumber();
```

Comment:
Why convert it `toNumber()` if it's later added to another BigNumber?

Reply:
Good catch! I used it for logging and forgot to remove it